app/main.py

```diff
@@ -1,6 +1,16 @@
-from typing import Callable
+from typing import Callable, Any
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cache_dict = {}
+
+    def inner(*args, **kwargs) -> Any:
+        if args not in cache_dict:
+            cache_dict.update({args: func(*args, **kwargs)})
```

Comment:
no need to use update method here. You can simply insert into dict by key. You should use update when you have multiple key/value pairs, want to merge dicts ect.
In this case you can choose simple inserting by key, it will also be more readable